databricks/koalas/series.py

```diff
@@ -4793,19 +4793,56 @@ def dot(self, other) -> Union[Scalar, "Series"]:
 
         >>> s @ s
         14
+
+        >>> kdf = ks.DataFrame({'x': [0, 1, 2, 3], 'y': [0, -1, -2, -3]})
+        >>> kdf
+           x  y
+        0  0  0
+        1  1 -1
+        2  2 -2
+        3  3 -3
+
+        >>> with ks.option_context("compute.ops_on_diff_frames", True):
+        ...     s.dot(kdf)
+        ...
+        x    14
+        y   -14
+        dtype: int64
         """
         if isinstance(other, DataFrame):
-            raise ValueError(
-                "Series.dot() is currently not supported with DataFrame since "
-                "it will cause expansive calculation as many as the number "
-                "of columns of DataFrame"
-            )
-        if self._kdf is not other._kdf:
-            if len(self.index) != len(other.index):
+            if not self.index.equals(other.index):
                 raise ValueError("matrices are not aligned")
-        if isinstance(other, Series):
+
+            combined = combine_frames(self._kdf, other)
+            sdf = combined._internal.spark_frame
+
+            this_data_spark_column_name = combined["this"]._internal.data_spark_column_names[0]
```

Comment:
Why can we use `data_spark_column_names[0]`?

Reply:
Since `this` is series, can't we assume it to be data_spark_column_names[0]?

Reply:
You are using `self._kdf` to combine frames. In that case, `data_spark_column_names[0]` is not always the column for the `Series`.

Please try:

```py
ks.DataFrame({'a': [1, 2, 3], 'b': [4, 5, 6]}).b.dot(ks.DataFrame({'c': [7, 8, 9]}))
```

Reply:
Makes sense, changed it to "to_frame" instead!